main_logic/core.py

```
@@ -630,7 +636,18 @@ async def handle_connection_error(self, message=None):
             else:
                 await self.send_status(json.dumps({"code": "API_UNKNOWN_ERROR", "details": {"msg": message_text}}))
         logger.info("💥 Session closed by API Server.")
-        await self.disconnected_by_server()
+        await self.disconnected_by_server(expected_session=expected_session)
+
+    def _bind_session_lifecycle_callbacks(self, session):
+        async def on_connection_error(message=None, session_ref=session):
+            await self.handle_connection_error(message, expected_session=session_ref)
+
+        setattr(session, 'on_connection_error', on_connection_error)
```

Ruff (0.15.4)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)